TLM/TLM/Manager/Impl/SpeedLimitManager.cs

```diff
@@ -48,19 +52,61 @@ public class SpeedLimitManager
         public static readonly SpeedLimitManager Instance = new();
 
         /// <summary>For each NetInfo name: custom speed limit.</summary>
-        private readonly Dictionary<string, float> customLaneSpeedLimit_ = new();
+        private readonly Dictionary<NetInfo, float> customNetinfoSpeedLimits_ = new();
+
+        /// <summary>determine if speed limits on this netinfo can be customised</summary>
+        public bool IsCustomisable(NetInfo netinfo) {
+            if (!netinfo) {
+                Log.Warning("Skipped NetINfo with null info");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(netinfo.name)) {
+                Log.Warning("Skipped NetINfo with empty name");
+                return false;
+            }
+
+            if (netinfo.m_netAI == null) {
+                Log.Warning($"Skipped NetInfo '{netinfo.name}' with null AI");
+                return false;
+            }
```

Comment:
It is only called when caching speed limits. 
It is also called by the tool but the tool will not pass a null/nameless netinfo.
If there is any other unforeseen scenario, then yes it will spam warnings.